### PR TITLE
[RADFISH-54] Setup storage functionality for offline data

### PR DESCRIPTION
**Pull request by melsoriano**
Create methods for storing offline data in `localStorage` or `indexedDB`.

**NOTE:** You will need to create a `.env.development.local` file. Then, copy and paste the example variables from `.env.example`

**Example usage:**

```javascript
const indexedDBMethod = new IndexedDBMethod();
const indexedDBStorageModel = StorageModelFactory.createModel(indexedDBMethod);

indexedDBStorageModel.saveEntry(formData);
```

### CODE REVIEW

File: src/storage/LocalStorageMethod.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { StorageMethod } from "./StorageMethod";
+
+export class LocalStorageMethod extends StorageMethod {
```
Comment (brockinit) [inline src/storage/LocalStorageMethod.js:3]: Can you add documentation to this file?

File: src/storage/IndexedDBMethod.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { generateUUID } from "../utilities";
+import { StorageMethod } from "./StorageMethod";
+import { db } from "./indexedDB";
+
+export class IndexedDBMethod extends StorageMethod {
```
Comment (brockinit) [inline src/storage/IndexedDBMethod.js:5]: Can you add docs to this file?

File: src/storage/indexedDB.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import Dexie from "dexie";
+
+export const db = new Dexie(process.env.REACT_APP_INDEXED_DB_NAME);
+
+db.version(process.env.REACT_APP_INDEXED_DB_VERSION).stores({
```
Comment (brockinit) [inline src/storage/indexedDB.js:5]: Can you add documentation explaining how to add support for additional fields?

File: src/utilities/cryptoWrapper.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export const generateUUID = () => {
```
Comment (brockinit) [inline src/utilities/cryptoWrapper.js:1]: Docs plz

**Review by brockinit — CHANGES_REQUESTED** [review]
Please add further documentation to make it clear to the end-developer how they can use/modify this functionality, if needed!

File: src/storage/LocalStorageMethod.js (hunk 1 above)
Comment (thgaskell) [inline src/storage/LocalStorageMethod.js:3]:
```suggestion
export class LocalStorageMethod extends StorageMethod {
  constructor (key) {
    this.key = key
  }
```

File: src/storage/StorageMethod.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+export class StorageMethod {
+  save(_data, _key) {
```
Comment (thgaskell) [inline src/storage/StorageMethod.js:2]:
```suggestion
  create(_data) {
```

File: src/storage/StorageMethod.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    throw new Error("Method not implemented.");
+  }
+
+  load(_key) {
```
Comment (thgaskell) [inline src/storage/StorageMethod.js:6]:
```suggestion
  find(_criteria) {
```

File: src/storage/StorageMethod.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    throw new Error("Method not implemented.");
+  }
+
+  loadOne(_uuid, _key) {
+    throw new Error("Method not implemented.");
+  }
+
+  editOne(_uuid, _data, _key) {
```
Comment (thgaskell) [inline src/storage/StorageMethod.js:14]:
```suggestion
  update(_criteria, _data) {
```

File: src/storage/StorageMethod.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+export class StorageMethod {
+  save(_data, _key) {
+    throw new Error("Method not implemented.");
+  }
+
+  load(_key) {
+    throw new Error("Method not implemented.");
+  }
+
+  loadOne(_uuid, _key) {
```
Comment (thgaskell) [inline src/storage/StorageMethod.js:10]: Let's remove this in favor of just `find`.

**Review by thgaskell — CHANGES_REQUESTED** [review]

File: src/storage/LocalStorageMethod.js (hunk 1 above)
Comment (melsoriano) [reply src/storage/LocalStorageMethod.js:4]: Will add docs in a separate PR

File: src/storage/IndexedDBMethod.js (hunk 2 above)
Comment (melsoriano) [reply src/storage/IndexedDBMethod.js:5]: Will add docs in a separate PR

File: src/utilities/cryptoWrapper.js (hunk 4 above)
Comment (melsoriano) [reply src/utilities/cryptoWrapper.js:1]: Will add docs in a separate PR

File: src/storage/LocalStorageMethod.js (hunk 1 above)
Comment (melsoriano) [reply src/storage/LocalStorageMethod.js:4]: Done ✅

File: src/storage/StorageMethod.js (hunk 6 above)
Comment (melsoriano) [reply src/storage/StorageMethod.js]: done ✅

File: src/storage/StorageMethod.js (hunk 8 above)
Comment (melsoriano) [reply src/storage/StorageMethod.js]: done ✅

**Pull request merged**